sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundle.java

```diff
@@ -0,0 +1,289 @@
+package org.sdase.commons.server.spring.data.mongo;
+
+import static java.util.Arrays.asList;
+
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.mongodb.ConnectionString;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import io.dropwizard.Configuration;
+import io.dropwizard.ConfiguredBundle;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import javax.validation.constraints.NotNull;
+import org.apache.commons.lang3.StringUtils;
+import org.sdase.commons.server.spring.data.mongo.converter.DateToZonedDateTimeConverter;
+import org.sdase.commons.server.spring.data.mongo.converter.LocalDateToStringConverter;
+import org.sdase.commons.server.spring.data.mongo.converter.StringToLocalDateConverter;
+import org.sdase.commons.server.spring.data.mongo.converter.StringToZonedDateTimeConverter;
+import org.sdase.commons.server.spring.data.mongo.converter.ZonedDateTimeToDateConverter;
+import org.sdase.commons.server.spring.data.mongo.health.MongoHealthCheck;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.mongodb.MongoDatabaseFactory;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.SimpleMongoClientDatabaseFactory;
+import org.springframework.data.mongodb.core.convert.DbRefResolver;
+import org.springframework.data.mongodb.core.convert.DefaultDbRefResolver;
+import org.springframework.data.mongodb.core.convert.MappingMongoConverter;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.data.mongodb.core.convert.MongoCustomConversions;
+import org.springframework.data.mongodb.core.mapping.MongoMappingContext;
+import org.springframework.data.mongodb.repository.support.MongoRepositoryFactoryBean;
+import org.springframework.data.repository.Repository;
+
+public class SpringDataMongoBundle<C extends Configuration> implements ConfiguredBundle<C> {
+
+  /**
+   * Adding our own JSR 310 converters to stay backwards compatible with the old Morphia Bundle.
+   * Spring usually provides {@link org.springframework.data.convert.Jsr310Converters} that are not
+   * used here intentionally.
+   */
+  private static final Set<Converter<?, ?>> DEFAULT_CONVERTERS =
+      new LinkedHashSet<>(
+          List.of(
+              // LocalDate
+              new LocalDateToStringConverter(),
+              new StringToLocalDateConverter(),
+              // ZonedDateTime
+              new DateToZonedDateTimeConverter(),
+              new StringToZonedDateTimeConverter(),
+              new ZonedDateTimeToDateConverter()));
```

Comment:
Should we introduce `enableMorphiaCompatibility()` in the builder so that we can use it for upgrades from v2 to v3 but use the Spring Data defaults for new services? Imho that would pay out in the long term.